bin-ranges-transfer/src/test/java/uk/gov/pay/java_lambdas/bin_ranges_transfer/util/UtilsTest.java

```diff
@@ -71,4 +89,21 @@ private static Stream<Arguments> extractDateString_shouldThrowException_whenDate
             Arguments.of("WP_341BIN_V04_20001035_001.CSV")
         );
     }
+
+    private static Stream<Arguments> extractVersion_shouldReturnVersion_whenPresent() {
+        return Stream.of(
+            Arguments.of("WP_341BIN_V04_20240212_001.CSV", Version.V04),
+            Arguments.of("WP_341BIN_V03_20240102_001.CSV", Version.V03),
+            Arguments.of("WP_341BIN_V01_20242311_001.CSV", Version.UNKNOWN),
+            Arguments.of("WP_341BIN_V99_20242311_001.CSV", Version.UNKNOWN)
+        );
+    }
+
+    private static Stream<Arguments> extractVersion_shouldThrowException_whenVersionIsMissingOrInvalidOrNotRecognised() {
+        return Stream.of(
+            Arguments.of("WP_341BIN_00000000_001.CSV", BIN_RANGE_VERSION_NOT_FOUND_ERROR_MESSAGE),
+            Arguments.of("WP_341BIN_VAA_20001035_001.CSV", BIN_RANGE_VERSION_NOT_FOUND_ERROR_MESSAGE)
+            
+        );
+    }
```

Comment:
Again, I think `@CsvSource` would work for this.

Reply:
we still need`@MethodSource` for these ones imo, I prefer keeping the error messages as constants up top and afaik you can't pass an enum using csvsource